### PR TITLE
create workspace using templates

### LINKED ISSUE #1102: Ask to include default set of files and folders for new workspace
add a checkbox to modal

### PR DESCRIPTION
**Pull request by Aniket-Engg**
fixes #1102 

Templates for workspace introduced. A template will help user to decide the starting point for a project.

Existing default workspace will still be the default choice for template. 2 new templates have been introduced for now: Blank & ERC20.

### CODE REVIEW

File: libs/remix-ui/workspace/src/lib/actions/events.ts [hunk 1]
```diff
@@ -10,8 +10,8 @@ let plugin, dispatch: React.Dispatch<any>
 export const listenOnPluginEvents = (filePanelPlugin) => {
   plugin = filePanelPlugin
 
-  plugin.on('filePanel', 'createWorkspaceReducerEvent', (name: string, isEmpty = false, cb: (err: Error, result?: string | number | boolean | Record<string, any>) => void) => {
-    createWorkspace(name, isEmpty, cb)
+  plugin.on('filePanel', 'createWorkspaceReducerEvent', (name: string, workspaceTemplateName, isEmpty = false, cb: (err: Error, result?: string | number | boolean | Record<string, any>) => void) => {
```
Comment (yann300) [inline libs/remix-ui/workspace/src/lib/actions/events.ts:13]: can `workspaceTemplateName` be typed ?

File: libs/remix-ui/workspace/src/lib/actions/workspace.ts [hunk 2]
```diff
@@ -41,17 +40,17 @@ export const addInputField = async (type: 'file' | 'folder', path: string, cb?:
   return promise
 }
 
-export const createWorkspace = async (workspaceName: string, isEmpty = false, cb?: (err: Error, result?: string | number | boolean | Record<string, any>) => void) => {
+export const createWorkspace = async (workspaceName: string, workspaceTemplateName, isEmpty = false, cb?: (err: Error, result?: string | number | boolean | Record<string, any>) => void) => {
```
Comment (yann300) [inline libs/remix-ui/workspace/src/lib/actions/workspace.ts:43]: `workspaceTemplateName` is a string ? can you put the type?

File: libs/remix-ui/workspace/src/lib/actions/workspace.ts [hunk 3]
```diff
@@ -150,15 +149,21 @@ export const loadWorkspacePreset = async (template: 'gist-template' | 'code-temp
...
-    case 'default-template':
-      // creates a new workspace and populates it with default project template.
-      // insert example contracts
-      for (const file in examples) {
-        try {
-          await workspaceProvider.set(examples[file].name, examples[file].content)
-        } catch (error) {
-          console.error(error)
+    default:
+      try {
```
Comment (yann300) [inline libs/remix-ui/workspace/src/lib/actions/workspace.ts:153]: if this somehow fails we should now probably use `displayNotification`

File: libs/remix-ui/workspace/src/lib/actions/workspace.ts (hunk 2 above)
Comment (Aniket-Engg) [reply libs/remix-ui/workspace/src/lib/actions/workspace.ts]: fixed

File: libs/remix-ui/workspace/src/lib/actions/workspace.ts (hunk 3 above)
Comment (Aniket-Engg) [reply libs/remix-ui/workspace/src/lib/actions/workspace.ts:153]: added

**Review by yann300 — APPROVED** [review]

**Pull request merged**